03_launch_mgmt_cluster.sh

```diff
@@ -468,7 +468,22 @@ fi
 if [ "${EPHEMERAL_CLUSTER}" != "tilt" ]; then
   patch_clusterctl
   launch_cluster_api_provider_metal3
-  apply_bm_hosts
 fi
 
 launch_ironic
+
+if [ "${EPHEMERAL_CLUSTER}" != "tilt" ]; then
+  if [[ "${BMO_RUN_LOCAL}" != true ]]; then
+    if ! kubectl rollout status deployment capm3-baremetal-operator-controller-manager -n capm3-system --timeout=5m; then
+      echo "baremetal-operator-controller-manager deployment can not be rollout"
+      exit 1
+    fi
+    #Wait servers are ready and certificates are injected.
+    sleep 30
```

Comment:
I am not a fan of sleep, is there a way to do some sort of a health check or a proper check to see if certificates are injected or not?

Reply:
Me neither. I think latency is between ca-injector and api server. Because I immediately see log in BMO `"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}`. After doing some research, we might need to sleep in here to give time `ca-injector` updates required places.

Reply:
Me neither. I think latency is between ca-injector and api server. Because I immediately see log in BMO `"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}`. After doing some research, we might need to sleep in here to give time `ca-injector` updates required places.

Reply:
I removed sleep for now.